### PR TITLE
add ng2-formly 2.0.0-alpha.5

### LINKED ISSUE #7568: [Request] Add ng2-formly
**Library name:** ng2-formly
**Git repository url:** https://github.com/formly-js/ng2-formly
**npm package url(optional):** https://www.npmjs.com/package/ng2-formly
**License(s):** MIT
**Official homepage:** http://formly-js.github.io/ng2-formly/
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by nolsherry**
@kennynaoh  , please help me to check this pull request, thank you vary much!
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/formly-js/ng2-formly
 * Official website (optional, not the repository): http://formly-js.github.io/ng2-formly/
 * NPM package url (optional): https://www.npmjs.com/package/ng2-formly
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 23
   - Count of watchers: 11
   - Count of forks: 3
 * NPM download stats (optional):
   - Downloads in the last day: 1
   - Downloads in the last week: 31
   - Downloads in the last month: 319

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [X] Auto-update target/source is valid.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

npm auto-update config added, close #7568, cc @mohammedzamakhan

### CODE REVIEW

**Comment by kennynaoh** [comment]
@PeterDaveHello 
Because the popularity of this PR is not enough
Please help me determine whether this PR could be added or not
Thanks

**Comment by PeterDaveHello** [comment]
We can add this lib.

**Comment by kennynaoh** [comment]
@nolsherry 
I think you used the wrong file
Maybe the correct file is in the directory of bundles

**Comment by mohammedzamakhan** [comment]
Yes the files are in bundles folder

**Comment by PeterDaveHello** [comment]
Thanks @mohammedzamakhan

**Comment by nolsherry** [comment]
@kennynaoh  Please help me to review this pull request, thank you!! :smile:

**Comment by kennynaoh** [comment]
@PeterDaveHello 
I think this PR is correct.
Please help me review it again.
Thanks.

File: ajax/libs/ng2-formly/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "name": "Zama Khan Mohammed",
+    "email": "mohammedzamakhan@gmail.com"
+  },
+  "npmName": "ng2-formly",
+  "npmFileMap": [
+    {
+      "basePath": "bundles",
+      "files": [
+        "ng2-formly.min.js",
+        "ng2-formly.js"
```
Comment (PeterDaveHello) [inline ajax/libs/ng2-formly/package.json]: Let's use `"ng2-formly*+.(js|map)"`

File: ajax/libs/ng2-formly/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  "homepage": "https://formly-js.github.io/ng2-formly/",
+  "dependencies": {
+    "angular2": "2.0.0-beta.14",
+    "es6-shim": "0.35.0",
+    "reflect-metadata": "0.1.2",
+    "rxjs": "5.0.0-beta.2",
+    "zone.js": "0.6.6",
+    "systemjs": "0.19.6"
+  },
+  "czConfig": {
```
Comment (PeterDaveHello) [inline ajax/libs/ng2-formly/package.json]: Please remove `czConfig`

**Comment by nolsherry** [comment]
@kennynaoh, please help me to review this pull request, thank you!

**Comment by kennynaoh** [comment]
@nolsherry 
Maybe you should explain which auto-update you use next time.
Thanks.
@PeterDaveHello 
I think this PR is correct.
Please help me review it again.
Thanks.

**Comment by PeterDaveHello** [comment]
The commit message should talk about the auto-update mechanism this time, let's update it.

**Comment by nolsherry** [comment]
@kennynaoh, please help me to review this pull request, thanks.

**Comment by kennynaoh** [comment]
@nolsherry 
Your commit message is too long.
Maybe you can take off the ```config```
Or put the auto-update message to third line.
Thanks.

**Comment by nolsherry** [comment]
@kennynaoh, please help me to review this pull request, thanks. :smiley:

**Comment by kennynaoh** [comment]
@PeterDaveHello 
I think this PR is correct.
Please help me review it again.
Thanks.

**Comment by PeterDaveHello** [comment]
Let's mention npm autoupdate in the first line.

**Comment by nolsherry** [comment]
@kennynaoh, please help me to review this pull request, thanks. :smile_cat:

**Comment by kennynaoh** [comment]
@PeterDaveHello 
I think this PR is correct.
Please help me review it again.
Thanks.

**Comment by PeterDaveHello** [comment]
hmmm ... both of you obviously didn't try to run the run the auto-update so you didn't notice that the config will not work.

**Comment by nolsherry** [comment]
@PeterDaveHello, Could I use this form
```**/*.+(!gz|js)``` 
in npm fileMap ?
I have used npm auto-update to comfirm this, thanks!

**Comment by nolsherry** [comment]
@PeterDaveHello, Could I use this form
```**/*.+(!gz|js) ``` 
in npm fileMap ?
I have used npm auto-update to comfirm this, thanks!

**Comment by PeterDaveHello** [comment]
@nolsherry You can just exclude the gz files.

**Comment by kennynaoh** [comment]
@nolsherry 
Maybe you can use this form.
```**/*.!(*gz) ```

**Comment by PeterDaveHello** [comment]
@kennynaoh have you run the auto-update tool with this pattern?

**Comment by kennynaoh** [comment]
@PeterDaveHello 
Yes, I have confirmed this.

**Comment by PeterDaveHello** [comment]
I think `"**/*.!(gz)"` is much better.

**Comment by nolsherry** [comment]
@PeterDaveHello, I try the form you suggest, it seems grep all files include  *.gz

**Comment by PeterDaveHello** [comment]
Type it too fast, this will work: `"**/!(*.gz)"`

**Comment by nolsherry** [comment]
@PeterDaveHello, thank you for your help, I have fixed this error!

**Comment by nolsherry** [comment]
@kennynaoh, please help me to review this pull request, thanks.

**Comment by kennynaoh** [comment]
@nolsherry 
Maybe rebasing to the latest master branch is much better.
Thanks.

**Comment by nolsherry** [comment]
@kennynaoh, please help me to review this pull request again, thanks.

**Comment by kennynaoh** [comment]
@PeterDaveHello 
I think this PR is correct.
Please help me review it again.
Thanks.

File: ajax/libs/ng2-formly/package.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "angular 2",
+    "angular 2 forms",
+    "json forms",
+    "form library"
+  ],
+  "license": "MIT",
+  "description": "angular2-formly is an Angular 2 module which has a Components to help customize and render JavaScript/JSON configured forms. The formly-form Component and the FormlyConfig service are very powerful and bring unmatched maintainability to your application's",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/formly-js/angular2-formly.git"
```
Comment (PeterDaveHello) [inline ajax/libs/ng2-formly/package.json]: Wrong repo url.

Comment (PeterDaveHello) [inline ajax/libs/ng2-formly/package.json]: @nolsherry please send a pull request to fix it.

**Comment by nolsherry** [comment]
@kennynaoh, I have fixed the wrong repo url, please help me to review this pull request again, thanks!

**Comment by PeterDaveHello** [comment]
> @nolsherry please send a pull request to fix it.

**Pull request merged**